addOns/pscan/src/main/java/org/zaproxy/addon/pscan/internal/scanner/PassiveScanTask.java

```diff
@@ -0,0 +1,230 @@
+                            && (scanner.appliesToHistoryType(hrefHistoryType)
+                                    || PassiveScanTaskHelper.getOptedInHistoryTypes()
+                                            .contains(hrefHistoryType))) {
+
+                        if (scanner instanceof PluginPassiveScanner) {
+                            PluginPassiveScanner pps = ((PluginPassiveScanner) scanner).copy();
+                            pps.setHelper(passiveScanData);
+                            scanner = pps;
+                        }
+                        pscanActionsSetter.accept(scanner);
+
+                        LOGGER.debug(
+                                "Running scan rule, URL {} plugin {}",
+                                msg.getRequestHeader().getURI(),
+                                scanner.getName());
+                        long scanRuleStartTime = System.currentTimeMillis();
+
+                        if (maxBodySize <= 0 || msg.getRequestBody().length() < maxBodySize) {
+                            scanner.scanHttpRequestSend(msg, href.getHistoryId());
+                            scanned = true;
+                        } else {
+                            Stats.incCounter("stats.pscan.reqBodyTooBig");
+                            LOGGER.debug(
+                                    "Request to {} body size {} larger than max configured {}",
+                                    msg.getRequestHeader().getURI(),
+                                    msg.getRequestBody().length(),
+                                    maxBodySize);
+                        }
+                        if (msg.isResponseFromTargetHost()) {
+                            if (maxBodySize <= 0 || msg.getResponseBody().length() < maxBodySize) {
+                                scanner.scanHttpResponseReceive(msg, href.getHistoryId(), src);
+                                scanned = true;
+                            } else {
+                                Stats.incCounter("stats.pscan.respBodyTooBig");
+                                LOGGER.debug(
+                                        "Response from {} body size {} larger than max configured {}",
+                                        msg.getRequestHeader().getURI(),
+                                        msg.getResponseBody().length(),
+                                        maxBodySize);
+                            }
+                        }
+                        if (scanned) {
+                            long timeTaken = System.currentTimeMillis() - scanRuleStartTime;
+                            if (scanner instanceof PluginPassiveScanner) {
+                                PluginPassiveScanner pps = (PluginPassiveScanner) scanner;
+                                Stats.incCounter(
+                                        "stats.pscan." + pps.getPluginId() + ".time", timeTaken);
+                            }
+                            // TODO remove at some point
```

Comment:
Why?

I think this has proven useful.

Reply:
The names are dynamic (e.g. i18n).

Reply:
Should we switch to ID?

Reply:
I'd be happy dropping the names and just keeping the IDs...

Reply:
Although for passive scanners (not plugin/scan rules) the name is hardcoded and those don't have an ID. Maybe worth keeping for those scanners.

Reply:
We don't need to block this PR. Can be a topic for another day 🙂